src/bittybuzz/src/BittyBuzzSystem.cpp

```diff
@@ -5,6 +5,7 @@ IUserInterface* BittyBuzzSystem::g_ui = NULL;
 const IBittyBuzzStringResolver* BittyBuzzSystem::g_stringResolver = NULL;
 IBittyBuzzClosureRegister* BittyBuzzSystem::g_closureRegister = NULL;
 IBittyBuzzMessageService* BittyBuzzSystem::g_messageService = NULL;
+IBSP* BittyBuzzSystem::g_bsp = NULL;
```

Comment:
This could be set with the container instead of setting it to null

Reply:
We should never use the container outside the main. Always inject the dependencies. If we use the container, we can't mock the BSP. The reason I use global variables here are because bittybuzz has no context, so the only way I can keep context is via global variables

Reply:
Could we mock the container? I just find it dangerous to have global pointers that are initialized as null... Also, the notation is a bit confusing, you write your variables BittyBuzzSystem::g_xxxx but since the variable is in the namespace, it is not global. 

Although, If we can't mock the container, then I would keep it as is since I prefer having unit tests on this compared to avoiding global pointers